test/middleware-test.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+var expect = require('chai').expect;
+var RSVP = require('rsvp');
+var broccoliMiddleware = require('./../lib/middleware');
+var fixture = require('./helpers/fixture-path');
+var TestHTTPServer = require('./helpers/test-http-server');
+
+describe('broccoli-middleware', function() {
+  describe('watcher resolves correctly', function() {
+    var watcher;
+    var server;
+
+    beforeEach(function() {
+      watcher = new RSVP.Promise(function(resolve, reject) {
+        resolve();
+      });
+
+      watcher['builder'] = {};
+    });
+
+    afterEach(function() {
+      server.stop();
+      server = null;
+    })
+
+    it('responds with the given file if file is on disk', function(done) {
+      watcher['builder']['outputPath'] = fixture('basic-file');
+      var middleware = broccoliMiddleware(watcher, {
+        autoIndex: false
+      });
+
+      server = new TestHTTPServer(middleware);
+
+      server.start()
+        .then(function(info) {
+          return server.request('/index.html', {
+            info: info
+          });
+        })
+        .then(function(content) {
+          expect(content).to.match(/This is broccoli middleware page/);
+          done();
```

Comment:
rather then using `done`, which if an assertion before it fails (like line 42) will cause the test to timeout. Lets use mocha's build in promise support.

Simply return a promise in the `it` `function`, like:

```js
it('foo', function() {
  return server.start().then(...);
});
```

If the promise rejects, the test fails, if the promise fulfills the test passes.